### DISCUSSION TITLE
[vitest-worker]: Timeout calling "fetch" with ...

### DISCUSSION THREAD

**Discussion by danLDev** [vitest-dev/vitest#6131] [Q&A]
We've recently upgraded from 1.6 to 2.0 and began seeing failures in the last couple of days with the following messages

`[vitest-worker]: Timeout calling "fetch" with "["/src/stores/index.ts","web"]"`
`[vitest-worker]: Timeout calling "resolveSnapshotPath"`
`[vitest-worker]: Timeout calling "onUnhandledError" with "[vitest-worker]: Timeout calling "onTaskUpdate""`

This happens on CI on every run and the file pointed to changes, so I do not believe the issue is localised to a given test/suite. It also only happens locally around 1/5 runs, so It's possibly resource related.

**Comment by jesse-r2c** [comment]
I'm having a similar issue. We have an Nx monorepo with multiple apps and libraries with the exact same configurations between them:

```ts
// vitest.base.ts - in root of the project
import { join, resolve } from "path";
import svgr from "vite-plugin-svgr";
import { defineConfig } from "vitest/config";
import { nxViteTsPaths } from "@nx/vite/plugins/nx-tsconfig-paths.plugin";

const config = defineConfig({
  plugins: [nxViteTsPaths(), svgr()],
  test: {
    setupFiles: join(__dirname, "/setupTests.ts"),
    globals: true,
    environment: "jsdom",
    alias: [
      {
        find: /^monaco-editor$/,
        replacement: "monaco-editor/esm/vs/editor/editor.api",
      },
    ],
  },
});

export default config;

```

In each app/library we have a `vitest.config.ts` file that extends the base config:

```ts
import { defineConfig, mergeConfig } from "vitest/config";

import baseConfig from "../../../vitest.base";

const config = defineConfig({
  root: __dirname,
  test: {
    include: ["./src/**/*.test.{ts,tsx}"],
    passWithNoTests: true,
    coverage: {
      reporter: ["text", "html", "clover", "json"],
      reportsDirectory: "../../../coverage/libs/shared/components",
    },
  },
});

export default mergeConfig(baseConfig, config);

```

When I run the tests I get the following error:
`Error: [vitest-worker]: Timeout calling "fetch" with "["path/to/libs/shared/components/src/lib/Preview/hooks/usePreview.test.ts","web"]"`

I've found that defining _exactly_ one worker fixes the issue, but it significantly slows down test runs since it can only run one test at a time. 

```ts
import { defineConfig, mergeConfig } from "vitest/config";

import baseConfig from "../../../vitest.base";

const config = defineConfig({
  root: __dirname,
  test: {
    include: ["./src/**/*.test.{ts,tsx}"],
    passWithNoTests: true,
    coverage: {
      reporter: ["text", "html", "clover", "json"],
      reportsDirectory: "../../../coverage/libs/shared/components",
    },
	minWorkers: 1, // 👈 hardcode minWorkers to 1
	maxWorkers: 1 // 👈 hardcode maxWorkers to 1
  },
});

export default mergeConfig(baseConfig, config);

```

There are a couple of cases that make this issue even stranger:
1. If I start a test run in watch mode and remove the `minWorkers` and `maxWorkers` parameters from the Vitest config, Vitest will pick up the config change and rerun the tests successfully with multiple workers. If I close the test runner and restart, it will fail again with the same configuration. 
2. One of our apps inexplicably works without `minWorkers` and `maxWorkers` set, but the config is _exaclty_ the same as every other config (except for the coverage path).

**Comment by jesse-r2c** [comment]
Update: In our case, switching to [SWC](https://github.com/vitejs/vite-plugin-react-swc) in our `vitest.base.ts` file seems to have fixed this issue.
